src/item_factory.cpp

```diff
@@ -414,6 +417,14 @@ void Item_factory::finalize_post( itype &obj )
             }
         }
     }
+
+    for( std::string &line : obj.ascii_picture ) {
+        if( utf8_width( remove_color_tags( line ) ) > ascii_art_width ) {
+            line = line.substr( 0, ascii_art_width );
```

Comment:
Trimming such a string is not as simple as that. One has to keep track of color tags (in order to reapply them after trimming), and consider each symbol individually (which can be longer than one byte).

I'd suggest leaving lines as they are, only issuing `debugmsg`. Then you can use `trim_and_print` to properly display those lines (if it isn't already used).

Reply:
So I used the code in `trim_and_print` but still in item_factory, is that an acceptable solution?

Reply:
Definitely not, the factory shan't print anything. I meant keeping `ascii_picture` untrimmed in the factory, but trimming it in `final_info`. `trim_and_print` is actually combined of two functions (as it's name clearly implies), so the first can be factored out and reused. There's a complication though: there are already two functions named `trim`, but they simply strip whitespaces, and neither can handle tags. So, a new descriptive name has to be found. `trim_by_length` maybe?

Reply:
Sorry I wasn't clear: I encapsulated the trimming part of trim_and_print in trim_and_return and used that in the factory

Reply:
Sorry, that's my bad actually: I should've taken a look at the code first. It's totally fine what you did. I'd only suggest getting rid of the `_and_return` part, and also swapping the arguments to get something like this:
```c++
std::string trim_by_length( const std::string  &text, int width );
```